File: engine/src/main/java/edu/austral/dissis/chess/engine/Board.java
package edu.austral.dissis.chess.engine;

public class Board {
  private int xCord;
  private int yCord;
  private Piece piece;

  public Board(int xCord, int yCord, Piece piece) {
    this.xCord = xCord;
    this.yCord = yCord;
    this.piece = piece;
  }
}

File: engine/src/main/kotlin/edu/austral/dissis/chess/engine/board/Board.kt
package edu.austral.dissis.chess.engine.board

import edu.austral.dissis.chess.engine.piece.Piece
import edu.austral.dissis.chess.engine.piece.Position

interface Board {

  fun getPiece(position: Position): Piece?

  fun getUsedPositions(): List<Position>

  fun movePiece(from: Position, to: Position): Board

  fun positionExists(position: Position): Boolean
}